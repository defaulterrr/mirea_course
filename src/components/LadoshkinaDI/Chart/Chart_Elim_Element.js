import React from 'react';
import BarChart from 'react-bar-chart';
// import './.css';
import styles from './Chart_CSS_Element.module.css';

const margin = {
  top: 20, right: 20, bottom: 30, left: 100,
};
const data = [
  { text: 'Men', value: 0.20 },
  { text: 'Women', value: 0.330 },
];

class Chart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      Men: null,
      Women: null,
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const { target } = event;
    const { value } = target;
    const { name } = target;

    this.setState({
      [name]: value,
    });
    if (name === 'Men') { data[0].value = value; }
    if (name === 'Women') { data[1].value = value; }

    console.log(data[0].value);
    console.log(data[1].value);
  }


  render() {
    return (
      <div className={styles.input}>
        <table>
          <tr>
            <td>
              <form>
                Men:
                <input
                  name="Men"
                  type="number"
                  value={this.state.Men}
                  onChange={this.handleInputChange}
                />
              </form>
            </td>
            <td>
              <form>
                Women:
                <input
                  name="Women"
                  type="number"
                  value={this.state.Women}
                  onChange={this.handleInputChange}
                />
              </form>
            </td>
          </tr>
        </table>
        <div>
          <BarChart
            ylabel="Quantity"
            data={data}
            width={700}
            height={500}
            margin={margin}
          />
        </div>
      </div>
    );
  }
}
export default Chart;
