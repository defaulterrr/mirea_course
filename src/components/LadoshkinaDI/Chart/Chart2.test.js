import React from 'react';
import renderer  from 'react-test-renderer';

import Chart from './Chart_Elim_Element';


it('Render correctly', () => {
const ChartComponent = renderer.create(<Chart />).toJSON();
expect(ChartComponent).toMatchSnapshot();
});