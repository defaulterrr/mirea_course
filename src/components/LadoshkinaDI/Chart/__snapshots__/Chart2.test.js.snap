// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly 1`] = `
<div
  className="input"
>
  <table>
    <tr>
      <td>
        <form>
          Men:
          <input
            name="Men"
            onChange={[Function]}
            type="number"
            value={null}
          />
        </form>
      </td>
      <td>
        <form>
          Women:
          <input
            name="Women"
            onChange={[Function]}
            type="number"
            value={null}
          />
        </form>
      </td>
    </tr>
  </table>
  <div>
    <svg>
      <g
        className="graph"
      />
    </svg>
  </div>
</div>
`;
